app/src/main/java/io/getstream/chat/android/core/poc/library/ChatClientImpl.kt

```diff
@@ -34,6 +36,46 @@ internal class ChatClientImpl constructor(
         }
     }
 
+    override fun setAnonymousUser(callback: (Result<ConnectionData>) -> Unit) {
+        socket.connect().enqueue { result ->
+
+            if (result.isSuccess) {
+                api.setConnection(result.data())
+            }
+
+            callback(result)
+        }
+    }
+
+    override fun setGuestUser(
+        user: User,
+        callback: (Result<ConnectionData>) -> Unit
+    ) {
+        anonymousApi.setGuestUser(
```

Comment:
That looks bad. Having almost the same instance of api just for one call look wrong.
Maybe there is way to do something like?
```
override fun setGuestUser(
        user: User,
        callback: (Result<ConnectionData>) -> Unit
) {
  api.anonimous = true
  socket.anonimous = true
  //...
}

override fun setUser(
        user: User,
        callback: (Result<ConnectionData>) -> Unit
) {
  api.anonimous = false
  socket.anonimous = true
  //...
}
```